[v14] Clarify deprecation schedule for {rpm,deb}.releases.teleport.dev (#36069)
gravitational/teleport: docs/pages/installation.mdx

repo at `https://deb.releases.teleport.dev/`, you can upgrade by
re-running the "Debian/Ubuntu (DEB)" install instructions above.

The legacy APT repo at `https://deb.releases.teleport.dev/` will not receive updates

> past Teleport 14 and will be turned off in after Teleport 14 is no longer supported.

Reviewer: ```suggestion
past Teleport 14 and will be turned off when Teleport 14 is no longer supported.
```

Author: Fixed in https://github.com/gravitational/teleport/pull/36069/commits/5e71e7eea19379adf39b0eb77d6e2c20d6cb48bc (though I only dropped the "in")
